### PR TITLE
changed assumptions in expr.__gt__

### PR DESCRIPTION
**Pull request by arihantparsoya**
this is fix for the issue #11122 . Changed the condition to evaluate the given expression if one side is real.
#### Previous output :

 `In [16]: x = Symbol('x', positive=False)`
`In [17]: x > 0`
`Out[17]: x > 0`
#### Output after changes in this PR:

 `In [16]: x = Symbol('x', positive=False)`
`In [17]: x > 0`
`Out[17]: False`
- Test is added.

### CODE REVIEW

**Comment by parsoyaarihant** [comment]
@asmeurer   can you review this?

**Comment by smichr** [comment]
-1

An `x` that is nonpositive could be `I` and `I > 0` is not a valid comparison so returning False for `var('x', positive=False)>0` is not valid.

**Comment by asmeurer** [comment]
@smichr doing things that way leads to extreme subtlety in using inequalities for assumptions. It basically means that you shouldn't ever use them to do assumptions checks, only `is_positive` (this change came up because of the `_value_check` function [here](https://github.com/sympy/sympy/blob/master/sympy/stats/crv_types.py), along with https://github.com/sympy/sympy/issues/11121).

The way this is now, `x.is_positive` is logically equivalent to `x > 0`. I think that `x not real` -> `not (x > 0)` is fine, as a "fact" in the assumptions. I believe the downside is that while facts like `(x > y)` -> `not (x < y)` are fine, you can't have the reverse `not (x < y)` -> `(x > y)`. It would have to be `not (x < y) and x real and y real` -> `(x > y)` (this is similar to the `infinite -> !finite` thing which was discussed at some point in the past). 

In other words, I think that assigning a truth value of false to `I > 0` is fine. The "paradox" is that `I < 0` is also false, but that's fine, because `>` and `<` are only opposites when one is true (similar to the "paradox" of `nan` or other non-numbers being neither finite nor infinite).

**Comment by smichr** [comment]
Thanks for the explanation. But the way it is with this PR, with 'x = var('x', positive=False)`, `x>0` evaluates but `I > 0` won't:

```python
>>> [(i.is_positive, i.is_nonnegative) for i in (x, I)]
[(False, None), (False, False)]
```

So I think the logic that return False in the one case will need to be tweaked to return False in the other case. I just don't want to see a divergence between symbol and constant behavior.

**Comment by parsoyaarihant** [comment]
I made the changes in the PR such that : 

```python 
>>> from sympy import *
>>> I > 0
False
>>> I < 0
False
```
But there are some problems with tests such as: 

```python 
raises(TypeError, lambda: oo < I)
```
